plugins/org.jboss.reddeer.workbench/src/org/jboss/reddeer/workbench/api/EditorFile.java

```diff
@@ -0,0 +1,54 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.reddeer.workbench.api;
+
+import java.io.InputStream;
+
+/**
+ * API for a file associated to an editor.
+ * 
+ * @author apodhrad
+ *
+ */
+public interface EditorFile {
+
+	/**
+	 * Returns an open input stream on the contents of this file. The client is
+	 * responsible for closing the stream when finished.
+	 * 
+	 * @return An open input stream
+	 */
+	InputStream getInputStream();
+
+	/**
+	 * Returns a path relative to the workspace.
+	 * 
+	 * @return Relative path
+	 */
+	String getRelativePath();
+
+	/**
+	 * Returns an absolute path.
+	 * 
+	 * @return Absolute path
+	 */
+	String getAbsolutePath();
+
+	/**
+	 * Evaluates an XPath expression and returns the result.
+	 * 
+	 * @param expression
+	 *            XPath expression
+	 * 
+	 * @return Result of the XPath expression
+	 */
+	String xpath(String expression);
```

Comment:
OK, finally get it ;) I went through code a bit and found out that we have already some xpath evaluation in org.jboss.reddeer.eclipse.wst.xml.ui.tabletree.XMLSourcePage, could you check it? It could be maybe reusable.

I suggest creation of following hierarchy:
We have
- Editor (interface we already have)
- AbstractEditor (abstract implementation of Editor)
- MultiPageEditor (with XML page source and xpath imlementation), TextEditor, ServerEditor, DefaultEditor (Implementations of AbstractEditor)

TODO:
- create interface XMLAssociatedEditor with default implementation of xpath method
- create XMLEditor which extends TextEditor and implements the XMLAssociatedEditor interface - there will go more fancy features over time
- add "implements" of the XMLAssociatedEditor to every editor where it is desired (for existing it is XMLSourcePage, and maybe those graphical editors)